pygmt/tests/test_grdcut_image.py

```diff
@@ -0,0 +1,78 @@
+"""
+Test pygmt.grdcut on images.
+"""
+
+import numpy as np
+import pytest
+import xarray as xr
+from pygmt import grdcut, which
+
+try:
+    import rioxarray
+
+    _HAS_RIOXARRAY = True
+except ImportError:
+    _HAS_RIOXARRAY = False
+
+
+@pytest.fixture(scope="module", name="region")
+def fixture_region():
+    """
+    Set the data region.
+    """
+    return [-53, -49, -20, -17]
+
+
+@pytest.fixture(scope="module", name="expected_image")
+def fixture_expected_image():
+    """
+    Load the expected grdcut image result.
+    """
+    return xr.DataArray(
+        data=np.array(
+            [
+                [[90, 93, 95, 90], [91, 90, 91, 91], [91, 90, 89, 90]],
+                [[87, 88, 88, 89], [88, 87, 86, 85], [90, 90, 89, 88]],
+                [[48, 49, 49, 45], [49, 48, 47, 45], [48, 47, 48, 46]],
+            ],
+            dtype=np.uint8,
+        ),
+        coords={
+            "band": [1, 2, 3],
```

Comment:
An off-topic question: 
```
In [3]: import rioxarray

In [4]: rioxarray.open_rasterio("earth_day_01d_p.tif")
Out[4]:
<xarray.DataArray (band: 3, y: 180, x: 360)> Size: 194kB
[194400 values with dtype=uint8]
Coordinates:
  * band         (band) int64 24B 1 2 3
  * x            (x) float64 3kB -179.5 -178.5 -177.5 ... 177.5 178.5 179.5
  * y            (y) float64 1kB 89.5 88.5 87.5 86.5 ... -86.5 -87.5 -88.5 -89.5
    spatial_ref  int64 8B 0
Attributes:
    AREA_OR_POINT:             Area
    STATISTICS_MAXIMUM:        255
    STATISTICS_MEAN:           49.507654320988
    STATISTICS_MINIMUM:        10
    STATISTICS_STDDEV:         65.299010755209
    STATISTICS_VALID_PERCENT:  100
    _FillValue:                0
    scale_factor:              1.0
    add_offset:                0.0
```
The `band` here is `[1, 2, 3]`, but `tile_map`, `band` is set to `[0, 1, 2]`:
 https://github.com/GenericMappingTools/pygmt/blob/41a0fe0a6a79849565348203737360855d40add8/pygmt/datasets/tile_map.py#L160

Are there any specific reasons for the differences or any conventions about `band`?

Reply:
Oh right, the 123 numbering comes from GDAL/rasterio. E.g. if you do `gdalinfo ~/.gmt/server/earth/earth_day/earth_day_01d_p.tif`, you'll get:

```
Driver: GTiff/GeoTIFF
Files: /home/user/.gmt/server/earth/earth_day/earth_day_01d_p.tif
Size is 360, 180
...
Band 1 Block=360x7 Type=Byte, ColorInterp=Undefined
  Min=10.000 Max=255.000 
  Minimum=10.000, Maximum=255.000, Mean=49.508, StdDev=65.299
  NoData Value=0
  Metadata:
    STATISTICS_MAXIMUM=255
    STATISTICS_MEAN=49.507654320988
    STATISTICS_MINIMUM=10
    STATISTICS_STDDEV=65.299010755209
    STATISTICS_VALID_PERCENT=100
Band 2 Block=360x7 Type=Byte, ColorInterp=Undefined
  Min=10.000 Max=255.000 
  Minimum=10.000, Maximum=255.000, Mean=49.911, StdDev=62.129
  NoData Value=0
  Metadata:
    STATISTICS_MAXIMUM=255
    STATISTICS_MEAN=49.910941358025
    STATISTICS_MINIMUM=10
    STATISTICS_STDDEV=62.129078727248
    STATISTICS_VALID_PERCENT=100
Band 3 Block=360x7 Type=Byte, ColorInterp=Undefined
  Min=10.000 Max=255.000 
  Minimum=10.000, Maximum=255.000, Mean=65.605, StdDev=44.820
  NoData Value=0
  Metadata:
    STATISTICS_MAXIMUM=255
    STATISTICS_MEAN=65.605154320988
    STATISTICS_MINIMUM=10
    STATISTICS_STDDEV=44.819618148349
    STATISTICS_VALID_PERCENT=100
```

> The `band` here is `[1, 2, 3]`, but `tile_map`, `band` is set to `[0, 1, 2]`:

Yeah, I simply forgot about GDAL's start from 1 convention when implementing https://github.com/GenericMappingTools/pygmt/pull/2125 :sweat_smile:. I'm guessing there's a historical reason for this, and I'm not sure if this is the exact reason why GDAL counts from 1, but optical satellite bands (e.g. from Landsat) are usually numbered Band 1, Band 2, etc, so that could be why? Unsure if we should re-number things, or just leave it.

Reply:
I think we should follow the GDAL convention and let band indexing start from 1.